SU2_CFD/src/fluid/CFluidScalar.cpp

```diff
@@ -54,7 +71,199 @@ CFluidScalar::CFluidScalar(su2double val_Cp, su2double val_gas_constant, const s
+  switch (config->GetKind_ConductivityModel()) {
+    case CONDUCTIVITYMODEL::CONSTANT:
+      for (int iVar = 0; iVar < n_species_mixture; iVar++) {
+        if (config->GetKind_ConductivityModel_Turb() == CONDUCTIVITYMODEL_TURB::CONSTANT_PRANDTL) {
+          ThermalConductivityPointers[iVar] = std::unique_ptr<CConstantConductivityRANS>(new CConstantConductivityRANS(
+              config->GetThermal_Conductivity_Constant(iVar), config->GetPrandtl_Turb(iVar)));
+        } else {
+          ThermalConductivityPointers[iVar] = std::unique_ptr<CConstantConductivity>(
+              new CConstantConductivity(config->GetThermal_Conductivity_Constant(iVar)));
+        }
+      }
+      break;
+    case CONDUCTIVITYMODEL::CONSTANT_PRANDTL:
+      for (int iVar = 0; iVar < n_species_mixture; iVar++) {
+        if (config->GetKind_ConductivityModel_Turb() == CONDUCTIVITYMODEL_TURB::CONSTANT_PRANDTL) {
+          ThermalConductivityPointers[iVar] = std::unique_ptr<CConstantPrandtlRANS>(
+              new CConstantPrandtlRANS(config->GetPrandtl_Lam(iVar), config->GetPrandtl_Turb(iVar)));
+        } else {
+          ThermalConductivityPointers[iVar] =
+              std::unique_ptr<CConstantPrandtl>(new CConstantPrandtl(config->GetPrandtl_Lam(iVar)));
+        }
+      }
+      break;
+    case CONDUCTIVITYMODEL::POLYNOMIAL:
+      for (int iVar = 0; iVar < n_species_mixture; iVar++) {
+        if (config->GetKind_ConductivityModel_Turb() == CONDUCTIVITYMODEL_TURB::CONSTANT_PRANDTL) {
+          ThermalConductivity = std::unique_ptr<CPolynomialConductivityRANS<N_POLY_COEFFS>>(
+              new CPolynomialConductivityRANS<N_POLY_COEFFS>(config->GetKt_PolyCoeffND(), config->GetPrandtl_Turb()));
+        } else {
+          ThermalConductivity = std::unique_ptr<CPolynomialConductivity<N_POLY_COEFFS>>(
+              new CPolynomialConductivity<N_POLY_COEFFS>(config->GetKt_PolyCoeffND()));
+        }
+      }
+      break;
+    default:
+      SU2_MPI::Error("Conductivity model not available.", CURRENT_FUNCTION);
+      break;
+  }
+}
+
+std::vector<su2double>& CFluidScalar::massToMoleFractions(const su2double* val_scalars) {
+  su2double mixtureMolarMass{0.0};
+  su2double val_scalars_sum{0.0};
+
+  for (int i_scalar = 0; i_scalar < n_species_mixture - 1; i_scalar++) {
+    massFractions[i_scalar] = val_scalars[i_scalar];
+    val_scalars_sum += val_scalars[i_scalar];
+  }
+  massFractions[n_species_mixture - 1] = 1 - val_scalars_sum;
+
+  for (int iVar = 0; iVar < n_species_mixture; iVar++) {
+    mixtureMolarMass += massFractions[iVar] / molarMasses[iVar];
+  }
+
+  for (int iVar = 0; iVar < n_species_mixture; iVar++) {
+    moleFractions.at(iVar) = (massFractions[iVar] / molarMasses[iVar]) / mixtureMolarMass;
+  }
+
+  return moleFractions;
+}
+
+su2double CFluidScalar::wilkeViscosity(const su2double* val_scalars) {
+  std::vector<su2double> phi;
+  std::vector<su2double> wilkeNumerator;
+  std::vector<su2double> wilkeDenumeratorSum;
```

Comment:
Not good for performance, use the MAX_SIZE and std::array

Reply:
done, thanks you so much for your suggestions